### PR TITLE
fix(gatsby-source-drupal): Verify nodes exist before looping through them

### LINKED ISSUE #22724: [gatsby-source-drupal] Cannot read property 'map' of undefined
## Description

I'm using the Gatsby Cloud live preview. The build compiles fine, and the preview functionality works. When creating a new Drupal node and Gatsby Cloud attempts to rebuild, it fails/crashes. A manual restart of the live preview works. 

This is a blocker since I can't expect the client to restart the live preview after each new Drupal node.

### Expected result

The build should complete, and the new Drupal node should be preview-able.

### Actual result

```
10:33:13 AM: TypeError: Cannot read property 'map' of undefined 
 - utils.js:148 handleWebhookUpdate [www]/[gatsby-source-drupal]/utils.js:148:70 
 - gatsby-node.js:217 [www]/[gatsby-source-drupal]/gatsby-node.js:217:20 
 - layer.js:95 Layer.handle [as handle_request] [www]/[express]/lib/router/layer.js:95:5 
 - index.js:317 trim_prefix [www]/[express]/lib/router/index.js:317:13 
 - index.js:284 [www]/[express]/lib/router/index.js:284:7 
 - index.js:335 Function.process_params [www]/[express]/lib/router/index.js:335:12 
 - index.js:275 next [www]/[express]/lib/router/index.js:275:10 
 - read.js:130 [www]/[body-parser]/lib/read.js:130:5
 - index.js:224 invokeCallback [www]/[raw-body]/index.js:224:16 
 - index.js:213 done [www]/[raw-body]/index.js:213:7 
 - index.js:273 IncomingMessage.onEnd [www]/[raw-body]/index.js:273:7
 - task_queues.js:84 processTicksAndRejections internal/process/task_queues.js:84:21
```

### Environment

I only see this on live preview, and not on my local.

### PR DESCRIPTION
**Pull request by smthomas**
## Description

There are certain types of entities in Drupal that are sent to the preview server that can not be processed. This adds a check to verify the nodes exist before trying to loop through and process the updates.

## Related Issues

Fixes #22724

### CODE REVIEW

File: packages/gatsby-source-drupal/src/utils.js [hunk 1]
```diff
@@ -147,10 +147,11 @@ const handleWebhookUpdate = async (
...
   } else {
     // if we are inserting new node, we need to update all referenced nodes
-    const newNodeReferencedNodes = referencedNodesLookup
-      .get(newNode)
-      .map(id => getNode(id))
-    nodesToUpdate.push(...newNodeReferencedNodes)
+    const newNodes = referencedNodesLookup.get(newNode)
+    if (typeof newNodes !== `undefined`) {
+      const newNodeReferencedNodes = newNodes.map(id => getNode(id))
+      nodesToUpdate.push(...newNodeReferencedNodes)
```
Comment (pvdz) [inline packages/gatsby-source-drupal/src/utils.js:153]: Can I suggest to just push in the loop instead? Up to you.

```suggestion
      newNodes.forEach(id => nodesToUpdate.push(getNode(id)))
```

**Comment by arshad** [comment]
We are using this patch and can confirm this works. This is good to go. LGTM.

File: packages/gatsby-source-drupal/src/utils.js [hunk 2]
```diff
@@ -147,10 +147,10 @@ const handleWebhookUpdate = async (
...
   } else {
     // if we are inserting new node, we need to update all referenced nodes
-    const newNodeReferencedNodes = referencedNodesLookup
-      .get(newNode)
-      .map(id => getNode(id))
-    nodesToUpdate.push(...newNodeReferencedNodes)
+    const newNodes = referencedNodesLookup.get(newNode)
+    if (typeof newNodes !== `undefined`) {
+      newNodes.forEach(id => nodesToUpdate.push(getNode(id)))
+    }
```
Comment (ascorbic) [inline packages/gatsby-source-drupal/src/utils.js:153]: You can use optional chaining for this:
```suggestion
    newNodes?.forEach(id => nodesToUpdate.push(getNode(id)))
```

**Comment by vladar** [comment]
Published in `gatsby-source-drupal@3.4.3`

**Pull request merged**